### PR TITLE
Refactors spiderlings into mobs

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
tweak: Spiderlings are now mobs. They'll now move about more regularly, can be sentienced/possessed, are affected by pest spray, and so on.
add: Spiderlings can now also be regular spiders, which don't grow into giant spiders.
/:cl:

I'm probably going to get a callout for putting a balance change in a refactor; i did it because otherwise i'd have to wait for this to be merged first. If it's an issue i can separate it.

The overall reasoning for the balance change is that 3 to 12 full spiders for every single egg cluster is a _bit_ much, and probably is the main reason that spiders are a huge round ender.

### CODE REVIEW

**Comment by DaxDupont** [comment]
Wait spiderlings are not mobs but **STRUCTURES**?

What fucking madlad devised this originally.

**Comment by cyclowns** [comment]
...they weren't mobs before?
welcome to ss13 code i guess

**Comment by Anonmare** [comment]
Sergenat Araneus, the HoS' pet, is a heavily edited bat.

It's all madness

**Comment by Buggy123** [comment]
# A T O M I C

**Comment by Doppeltongue** [comment]
Can spiderlings still be eaten?

**Comment by XDTM** [comment]
They stil drop the eatable version yes

File: code/modules/mob/living/simple_animal/friendly/spider.dm [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	var/amount_grown = 0
+	var/player_controlled = FALSE
+	var/grown_poison_type = "toxin"
+	var/grown_poison_per_bite = 5
+	var/directive = "" //Message from the mother
+	gold_core_spawnable = NO_SPAWN
+
+/mob/living/simple_animal/spiderling/giant/Life()
+	..()
+	if(isturf(loc))
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/friendly/spider.dm]: is this check still needed?

Comment (XDTM) [reply code/modules/mob/living/simple_animal/friendly/spider.dm]: I'm not 100% sure, but i figured that without it you'd get spiders growing in vents.

**Review by ChangelingRain — CHANGES_REQUESTED** [review]
Split the balance change out from the refactor, please.

File: code/modules/mob/living/simple_animal/friendly/spider.dm [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		return
+	forceMove(get_turf(exit_vent))
+	travelling_in_vent = FALSE
+	entry_vent = null
+	exit_vent = null
+	var/area/new_area = get_area(src)
+	if(new_area)
+		new_area.Entered(src)
+
+/mob/living/simple_animal/spiderling/Life()
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/friendly/spider.dm]: Should be in handle automated action/movement.

**Review by AnturK — CHANGES_REQUESTED** [review]

**Comment by XDTM** [comment]
Reviews addressed

File: code/modules/mob/living/simple_animal/friendly/spider.dm [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	if(!exit_vent || exit_vent.welded)
+		forceMove(entry_vent)
+		entry_vent = null
+		travelling_in_vent = FALSE
+		return
+	forceMove(get_turf(exit_vent))
+	travelling_in_vent = FALSE
+	entry_vent = null
+	exit_vent = null
+	var/area/new_area = get_area(src)
```
Comment (optimumtact) [inline code/modules/mob/living/simple_animal/friendly/spider.dm]: uh, forceMove will definitely be handling calling Entered on the area

Comment (optimumtact) [reply code/modules/mob/living/simple_animal/friendly/spider.dm]: if it's not, that's a bug

Comment (XDTM) [reply code/modules/mob/living/simple_animal/friendly/spider.dm]: It's old code, maybe it used to set loc directly. I'll remove that part then.

File: code/modules/mob/living/simple_animal/friendly/spider.dm [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+/mob/living/simple_animal/spiderling/proc/enter_vent()
+	travelling_in_vent = TRUE
+	forceMove(entry_vent)
+	var/travel_time = round(get_dist(loc, exit_vent.loc) / 2)
+	addtimer(CALLBACK(src, .proc/travel_vent), travel_time)
+
+/mob/living/simple_animal/spiderling/proc/travel_vent()
+	if(!exit_vent || exit_vent.welded)
+		forceMove(entry_vent)
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/friendly/spider.dm]: Shouldn't these exit to turf on failure ?

File: code/modules/mob/living/simple_animal/friendly/spider.dm (hunk 3 above)
Comment (optimumtact) [reply code/modules/mob/living/simple_animal/friendly/spider.dm]: okay, tanks

**Comment by AnturK** [comment]
What's the state of this.

**Pull request closed without merging**